StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1100CodeFixProvider.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace StyleCop.Analyzers.ReadabilityRules
+{
+    using System.Collections.Immutable;
+    using System.Composition;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CodeFixes;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+    [ExportCodeFixProvider(nameof(SA1100DoNotPrefixCallsWithBaseUnlessLocalImplementationExists), LanguageNames.CSharp)]
+    [Shared]
+    public class SA1100CodeFixProvider : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> _fixableDiagnostics =
+    ImmutableArray.Create(SA1110OpeningParenthesisMustBeOnDeclarationLine.DiagnosticId);
+
+        public override ImmutableArray<string> GetFixableDiagnosticIds()
+        {
+            return _fixableDiagnostics;
+        }
+
+        public override async Task ComputeFixesAsync(CodeFixContext context)
+        {
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                if (!diagnostic.Id.Equals(SA1100DoNotPrefixCallsWithBaseUnlessLocalImplementationExists.DiagnosticId))
+                    continue;
+
+                var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+                var node = root.FindNode(diagnostic.Location.SourceSpan) as BaseExpressionSyntax;
+                if (node == null)
+                {
+                    return;
+                }
+
+                var thisExpressionSyntax = SyntaxFactory.ThisExpression();
```

Comment:
Example:
```
#if true
        base
#else
    this
#endif
.ToString();
```
When I look at the syntax tree for this would I think will happen is that your code fix will remove the #if directive, some whitespaces and the end of line token. If you think that no one will ever do something incredible like this have a look at Json.Net. They love their #ifs and #elses